sentry-types/src/protocol/monitor.rs

```diff
@@ -0,0 +1,113 @@
+    /// Check-in failed or otherwise had some issues.
+    Error,
+    /// Check-in is expectred to complete.
+    InProgress,
+    /// Monitor did not check in on time.
+    Missed,
+    /// No status was passed.
+    #[serde(other)]
+    Unknown,
+}
+
+/// Configuration object of the monitor schedule.
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+#[serde(tag = "type")]
+pub enum Schedule {
+    /// A Crontab schedule allows you to use a standard UNIX crontab style schedule string to
+    /// configure when a monitor check-in will be expected on Sentry.
+    Crontab {
+        /// The crontab syntax string defining the schedule.
+        value: String,
+    },
+    /// A Interval scheudle allows you to configure a periodic checkin, that will occur at an
+    /// interval after the most recent check-in.
+    Interval {
+        /// The interval value.
+        value: u64,
+        /// The interval unit of the value.
+        unit: IntervalUnit,
+    },
+}
+
+/// The unit for the interval schedule type
+#[derive(Clone, Copy, Debug, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub enum IntervalUnit {
+    /// Year Interval.
+    Year,
+    /// Month Interval.
+    Month,
+    /// Week Interval.
+    Week,
+    /// Day Interval.
+    Day,
+    /// Hour Interval.
+    Hour,
+    /// Minute Interval.
+    Minute,
+}
+
+/// The monitor configuration playload for upserting monitors during check-in
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct MonitorConfig {
+    /// The monitor schedule configuration.
+    pub schedule: Schedule,
+
+    /// How long (in minutes) after the expected checkin time will we wait until we consider the
+    /// checkin to have been missed.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub checkin_margin: Option<u64>,
+
+    /// How long (in minutes) is the checkin allowed to run for in in_rogress before it is
+    /// considered failed.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub max_runtime: Option<u64>,
+
+    /// tz database style timezone string
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub timezone: Option<String>,
+}
+
+fn serialize_id<S: Serializer>(uuid: &Uuid, serializer: S) -> Result<S::Ok, S::Error> {
+    serializer.serialize_some(&uuid.as_simple())
+}
+
+/// The monitor check-in payload.
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct MonitorCheckIn {
+    /// Unique identifier of this check-in.
+    #[serde(serialize_with = "serialize_id")]
+    pub check_in_id: Uuid,
+
+    /// Identifier of the monitor for this check-in.
+    pub monitor_slug: String,
+
+    /// Status of this check-in. Defaults to `"unknown"`.
+    pub status: CheckInStatus,
+
+    /// The environment to associate the check-in with.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub environment: Option<String>,
```

Comment:
just curious as I know nothing about the crons product: do we not need/want a `release` as well?

Reply:
It's not something we support yet, or afaik have a plan to support